### ISSUE TITLE
Versions props linter FR: be more lax about `group:*` dependencies

### ISSUE THREAD

**Issue by JacekLach** [palantir/gradle-baseline#369]
Occasionally a product wants to keep a dependency that's not in the used BOM at the same version as another dependency in the BOM, for example you may have:

```
    versions.props: org.slf4j:* -> 1.7.25
    bom:
                    org.slf4j:jul-to-slf4j -> 1.7.25
                    org.slf4j:slf4j-api -> 1.7.25
```

because your product also depends on `org.slf4j:log4j-over-slf4j`.

You could manually specify the version of `org.slf4j:log4j-over-slf4j` in version.props, but then it's difficult to track bumps in the BOM.

You could `lint:OFF` the `org.slf4j:*` version in your version.props but then if the BOM upgrades you will force-override the version.

What we'd like is to see an error if `org.slf4j:*` is set to a version different than the BOM version,  but not see any errors or warnings when the versions match. Best if you don't have to flag it via comment, plugin just learns that a `*` dependency has special meaning.

**Comment by rubenfiszel** [comment]
Should be fixed by #371

**Comment by JacekLach** [comment]
Yep, seems fixed
